tests/ut/test_switch_ut.py

```diff
@@ -11,10 +12,89 @@ def skip_all(testbed_instance):
         pytest.skip("invalid for \"{}\" testbed".format(testbed.name))
 
 
-@pytest.mark.parametrize(
-    "attr,attr_type",
-    switch_attrs
-)
-def test_get_attr(npu, dataplane, attr, attr_type):
-    status, data = npu.get_by_type(npu.switch_oid, attr, attr_type, do_assert = False)
-    npu.assert_status_success(status)
+class TestSwitch:
+    state = dict()
+    miss_packet_actions_map = tuple(itertools.product(
+    (
+        "SAI_SWITCH_ATTR_FDB_UNICAST_MISS_PACKET_ACTION",
+        "SAI_SWITCH_ATTR_FDB_BROADCAST_MISS_PACKET_ACTION",
+        "SAI_SWITCH_ATTR_FDB_MULTICAST_MISS_PACKET_ACTION"
+    ),
+    (
+        "SAI_PACKET_ACTION_DROP",
+        "SAI_PACKET_ACTION_FORWARD",
+        "SAI_PACKET_ACTION_COPY",
+        "SAI_PACKET_ACTION_COPY_CANCEL",
+        "SAI_PACKET_ACTION_TRAP",
+        "SAI_PACKET_ACTION_LOG",
+        "SAI_PACKET_ACTION_DENY",
+        "SAI_PACKET_ACTION_TRANSIT",
+        "SAI_PACKET_ACTION_DONOTDROP"
+    )))
+
+    @pytest.mark.parametrize(
+        "attr,attr_type",
+        switch_attrs
+    )
+    def test_get_attr(self, npu, dataplane, attr, attr_type):
+        status, data = npu.get_by_type(npu.switch_oid, attr, attr_type, do_assert = False)
+        npu.assert_status_success(status)
+
+
+    @pytest.mark.parametrize(
+            "attr,attr_value",
+            [
+                ("SAI_SWITCH_ATTR_SRC_MAC_ADDRESS", "00:11:22:33:44:55"),
+                ("SAI_SWITCH_ATTR_SRC_MAC_ADDRESS", "00:00:00:00:00:00"),
+                ("SAI_SWITCH_ATTR_SWITCHING_MODE", "SAI_SWITCH_SWITCHING_MODE_CUT_THROUGH"),
+                ("SAI_SWITCH_ATTR_SWITCHING_MODE", "SAI_SWITCH_SWITCHING_MODE_STORE_AND_FORWARD"),
+                ("SAI_SWITCH_ATTR_RESTART_WARM", "true"),
+                ("SAI_SWITCH_ATTR_RESTART_WARM", "false"),
+                ("SAI_SWITCH_ATTR_WARM_RECOVER", "true"),
+                ("SAI_SWITCH_ATTR_WARM_RECOVER", "false"),
+                ("SAI_SWITCH_ATTR_COUNTER_REFRESH_INTERVAL", "0"),
+                ("SAI_SWITCH_ATTR_COUNTER_REFRESH_INTERVAL", "1"),
+                ("SAI_SWITCH_ATTR_SWITCH_SHELL_ENABLE", "true"),
+                ("SAI_SWITCH_ATTR_SWITCH_SHELL_ENABLE", "false"),
+                ("SAI_SWITCH_ATTR_FAST_API_ENABLE", "true"),
+                ("SAI_SWITCH_ATTR_FAST_API_ENABLE", "false"),
+                ("SAI_SWITCH_ATTR_MIRROR_TC", "255"),
+                ("SAI_SWITCH_ATTR_MIRROR_TC", "1"),
+                ("SAI_SWITCH_ATTR_CRC_CHECK_ENABLE", "true"),
+                ("SAI_SWITCH_ATTR_CRC_CHECK_ENABLE", "false"),
+                ("SAI_SWITCH_ATTR_CRC_RECALCULATION_ENABLE", "true"),
+                ("SAI_SWITCH_ATTR_CRC_RECALCULATION_ENABLE", "false"),
+            ],
+        )
+    def test_set_switch_attr(self, npu, dataplane, attr, attr_value):
+        if attr_value is None:
```

Comment:
This approach is used to convert an attribute to the default state after it was modified. E.g., https://github.com/opencomputeproject/SAI-Challenger/blob/b7ef8dc28fd56abf91d6284b2e90eca108869b57/tests/ut/test_vrf_ut.py#L32

To do that, the cache (state dict) should be initialized first. E.g., https://github.com/opencomputeproject/SAI-Challenger/blob/b7ef8dc28fd56abf91d6284b2e90eca108869b57/tests/ut/test_vrf_ut.py#L25

Please fix

Reply:
Fixed.